### PR TITLE
[OneBot] fix Sending message failed but still returns success

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ishkong** [comment]
retcode改成-1000吧，纪念一下死去的酷Q（x
-1000表示发生未知错误（你应该看日志了

File: Lagrange.OneBot/Core/Operation/Message/SendGroupForwardOperation.cs [hunk 1]
```diff
@@ -22,10 +22,13 @@ public async Task<OneBotResult> HandleOperation(BotContext context, JsonNode? pa
...
-            var ret = await context.SendMessage(chain);
-            int hash = MessageRecord.CalcMessageHash(chain.MessageId, ret.Sequence ?? 0);
+            var result = await context.SendMessage(chain);
 
-            return new OneBotResult(new OneBotForwardResponse(hash, multi.ResId ?? ""), (int)ret.Result, "ok");
+            if (!result.Sequence.HasValue || result.Sequence.Value == 0) return new OneBotResult(null, -1, "failed");
+
+            int hash = MessageRecord.CalcMessageHash(chain.MessageId, result.Sequence.Value);
+
+            return new OneBotResult(new OneBotForwardResponse(hash, multi.ResId ?? ""), (int)result.Result, "ok");
```
Comment (ishkong) [inline Lagrange.OneBot/Core/Operation/Message/SendGroupForwardOperation.cs:31]: 这里retcode是不是该为0

https://github.com/botuniverse/onebot-11/blob/d4456ee706f9ada9c2dfde56a2bcfc69752600e4/communication/http.md?plain=1#L61

**Review by ishkong — CHANGES_REQUESTED** [review]

**Comment by Executor-Cheng** [comment]
由于 Author 注销账号, 暂时关闭此 PR

**Review by CwkDark — APPROVED** [review]

**Review by ishkong — APPROVED** [review]

**Comment by KusaGumina** [comment]
Unsigned,Close

**Pull request closed without merging**